.github/workflows/ci.yml

```diff
@@ -111,17 +188,29 @@ jobs:
           retention-days: 5
 
   unit-tests:
-    needs: [check-permissions]
+    needs: [check-permissions, build-packages]
+    if: ${{ github.event_name != 'pull_request' || github.event.pull_request.draft == false }}
     name: Unit Tests
+    permissions:
```

Comment:
Is this a new requirement? The previous workflow didn't have these permissions, but artifacts were uploaded (as far as I know that is)

Reply:
These are more restrictive then default token permissions so they are an additional layer of locking things down. I am fine if we want to skip it though

Reply:
oh no in that case let's leave them! I thought they were adding previously unavailable permissions